tools/c7n_azure/c7n_azure/resources/vnet.py

```diff
@@ -32,4 +34,51 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.network'
         client = 'NetworkManagementClient'
         enum_spec = ('virtual_networks', 'list_all', None)
+        default_report_fields = (
+            'name',
+            'location',
+            'resourceGroup'
+        )
         resource_type = 'Microsoft.Network/virtualNetworks'
+
+@Vnet.filter_registry.register('gw-without-ipsec-policies')
+class IPSecAlgorithmFilter(ValueFilter):
+    """Virtual Networks Gateway Resource IPSec configured
+
+    :example:
+
+    This set of policies will find all Virtual Networks that are not
+      configured with a cryptygraphic algorithm.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gateway-configured-with-cryptographic-algorithm
+            resource: azure.vnet
+            filters: 
+              -type: configured-with-cryptographic-algorithm,
+    """
+    schema = type_schema('virtual_networks', rinherit=ValueFilter.schema)
+    schema_alias = False
+    AnnotationKey = "c7n:matched-vpn"
+    AnnotationKeyConn = "c7n:matched-conn"
+      
+    def process(self, resources, event=None):
+      client = self.manager.get_client()
+      matched = []
+      for resource in resources:
+          rg = ResourceIdParser.get_resource_group(resource["id"])
+          vpns = client.virtual_network_gateways.list(rg)
+          for vpn in vpns:
+              if vpn.gateway_type != "ExpressRoute":
+                vpnrg = ResourceIdParser.get_resource_group(vpn.id)
+                connections = client.virtual_network_gateway_connections.list(
+                  vpnrg
+                )
+                for connection in connections:
+                  if not connection.ipsec_policies:
+                      resource[self.AnnotationKey] = vpn.name
+                      resource[self.AnnotationKeyConn] = connection.name
+                      matched.append(resource)
```

Comment:
multiple vpns matching would cause the resource to be duplicated ins results. the annotation is stomping on previous matches.